FtcRobotController/src/main/java/com/qualcomm/ftcrobotcontroller/opmodes/PacmanBotHardwareBase.java

```diff
@@ -0,0 +1,142 @@
+    double driveMultiplier=1.0;          //Then the forward drive value is scaled to this.
+    double turnMultiplier=1.0;         //The turn value is scaled to this.
+    double motorExponent=1.0;           //The final motor powers are exponentially curved to this.
+    double finalRateMultiplier=1.0;     //The final motor powers are scaled to this.
+
+    public void setDriveClamp(double value){driveClamp=value;}
+    public void setTurnClamp(double value){turnClamp=value;}
+    public void setDriveExponent(double value){driveExponent=value;}
+    public void setTurnExponent(double value){turnExponent=value;}
+    public void setDriveMultiplier(double value){driveMultiplier=value;}
+    public void setTurnMultiplier(double value){turnMultiplier=value;}
+    public void setMotorExponent(double value){motorExponent=value;}
+    public void setFinalRateMultiplier(double value){finalRateMultiplier=value;}
+
+    public void setBrushPower(double power){brush.setPower(power);}
+
+    public ColorDetected getEyeColor(){
+        double red = eye.red();
+        double green = eye.green();
+        double blue = eye.blue();
+
+        if (Math.abs(red - blue)>COLOR_DETECTION_THRESHOLD) {
+            if (red > blue) {return ColorDetected.COLOR_RED;}
+            else {return ColorDetected.COLOR_BLUE;}
+        }
+        return ColorDetected.COLOR_NEITHER;
+    }
+
+    public String getColorString(ColorDetected color) {
+        switch (color) {
+            case COLOR_BLUE: return "Blue";
+            case COLOR_RED: return "Red";
+            case COLOR_NEITHER: return "Neither";
+        }
+        return "Error";
+    }
+
+    public void setEyeLED(boolean active) {
+        eye.enableLed(active);
+    }
+
+    public void drive(double drive_rate, double turn_rate) {
+        drive_rate = limit(drive_rate,-driveClamp,driveClamp);
+        turn_rate = limit(turn_rate, -turnClamp, turnClamp);
+        drive_rate = exp(drive_rate, driveExponent);
+        turn_rate = exp(turn_rate,turnExponent);
+        drive_rate = drive_rate * driveMultiplier;
+        turn_rate = turn_rate * turnMultiplier;
+        double motorLeftPower = limit(drive_rate + turn_rate,-1,1);
+        double motorRightPower = limit(drive_rate - turn_rate,-1,1);
+        motorLeftPower = exp(motorLeftPower,motorExponent) * finalRateMultiplier;
+        motorRightPower = exp(motorRightPower,motorExponent) * finalRateMultiplier;
+        telemetry.addData("Left",motorLeftPower);
+        telemetry.addData("Right",motorRightPower);
+        frontLeft.setPower(motorLeftPower);
+        frontRight.setPower(motorRightPower);
+        rearLeft.setPower(motorLeftPower * REAR_MULTIPLIER);
+        rearRight.setPower(motorRightPower * REAR_MULTIPLIER);
+    }
+    public double limit(double value,double min,double max){
+        if (value>max) {return max;}
+        if (value<min) {return min;}
+        return value;
+    }
+    public double exp(double value, double exponent){
```

Comment:
Logic was not right. Fix as

return Math.signum(value) * Math.pow(Math.abs(value), exponent)